hibernate-core/src/main/java/org/hibernate/query/sql/internal/ParameterParser.java

```diff
@@ -146,7 +146,7 @@ else if ( c == ':' ) {
 					recognizer.namedParameter( param, indx );
 					indx = chopLocation - 1;
 				}
-				else if ( c == '?' ) {
+				else if ( c == '?' && !recognizer.nativeJdbcParametersIgnored() ) {
```

Comment:
I think the check for the flag should rather move down to line 165, otherwise you'd also disable JPA positional parameter parsing with this config, not just JDBC parameters.

```suggestion
				else if ( c == '?' ) {
```

Reply:
Does JPA allow mixing positional and named parameters?